### ISSUE TITLE
[Internship Project]: Febie Lin

### ISSUE THREAD

**Issue by GemmaTuron** [ersilia-os/ersilia#711]
### Summary

Hello,

This is a public issue for a virtual daily stand-up. We will use this to briefly share the tasks of the day and the challenges and advances made, so that we can ensure smooth support from the Ersilia mentors and alignment between daily tasks and overall internship goals.

### Scope

Initiative 🐋

### Objective(s)

Internship goals:

- Learn how to work in an Open Source Community
- Improve the infrastructure of the Ersilia Model Hub
- Improve the usability and platform compatibility of Ersilia Models, mainly via Docker Images
- Learn about AI/ML models for drug discovery
- Identify new relevant AI/ML models and datasets and add them to this [discussion](https://github.com/ersilia-os/ersilia/discussions/694)
- Incorporate new AI/ML models in the Hub
- Improve the documentation of the Ersilia Model Hub for users
- Improve the documentation of the Ersilia Model Hub for developers

### Team

| Role & Responsibility | Username(s) | 
|--|--|
| Intern | @febielin |
| Mentor | @miquelduranfrigola  |
| Coordinator | @GemmaTuron |

### Timeline

Before starting your work, line up a few tasks and short description. This should not take long. For example, it could be something like:
Wednesday 21st June
- [ ] Create Intern daily task tracker
- [ ] Update the Ersilia GitHub Project
- [ ] Make sure Git Actions start upon success of the other workflows
- [ ] Work on the documentation for model testing
- [ ] Attend the Drug Discovery intro by WCAIR

**Comment by GemmaTuron** [comment]
In this example, following the tasks of Wednesday 21st June:

- [x]  Create Intern daily task tracker
- [x]  Update the Ersilia GitHub Project
- [x]  Make sure Git Actions start upon success of the other workflows
- [ ]  Work on the documentation for model testing
- [x]  Attend the Drug Discovery intro by WCAIR

I have created all the templates for the Interns, spend a couple of hours revising the [GitHub Project](https://github.com/orgs/ersilia-os/projects/1) and updating the tasks. I have been working on identifying the bug in the GitActions, solved in this [issue](https://github.com/ersilia-os/eos-template/issues/29).
I have set a meeting with @miquelduranfrigola to discuss in detail his comments on the [Model testing discussion](https://github.com/ersilia-os/ersilia/discussions/700) but I haven't been able to start writing yet, i am to do so by the end of the week.
Looking forward to the WCAIR next lesson!

Pro tip: adding the links to the issues and discussions you mention will be very helpful!

**Comment by febielin** [comment]
Wednesday, June 21, 2023:

 - [ ] Read and understand [Contribute models](https://ersilia.gitbook.io/ersilia-book/ersilia-model-hub/contribute-models), linked by Miquel this morning in the Slack. 
 - [ ] Attend RightsCon Takeaways event, hosted by Miquel 8-9 AM
 - [ ] Work on [Clean UP & Dockerization eos2v11](https://github.com/ersilia-os/eos2v11/issues/1)
 - [ ] See through the final steps of the [Demo Malaria Model](https://github.com/ersilia-os/ersilia/issues/709) to ensure understanding of the model incorporation flow
 - [ ] Testing of new models

**Comment by GemmaTuron** [comment]
Thansk @febielin  for the update! Good start and there is a lot to catch up, you'll get there :)

**Comment by febielin** [comment]
Thursday, June 22, 2023: 

- [x] Lab Meeting 8-9 AM
- [x] Meeting with Miquel 9-9:30 AM
- [ ] Test of new models 
- [ ] Testing, Cleanup, and Dockerization of [eos2v11](https://github.com/ersilia-os/eos2v11/issues/1)

**Comment by GemmaTuron** [comment]
great summary @febielin  thanks!

**Comment by febielin** [comment]
Monday, June 26, 2023:

- [ ] Refactorization of models
- [ ] Model testing
- [ ] If time, prep for Thursday presentation

Had some pesky internet troubles this morning, but that was resolved.

**Comment by febielin** [comment]
- [ ] Weekly Meeting 8-9 AM
- [ ] Meeting with Miquel 9-9:30 AM
- [ ] Refactorization of eos1af5
- [ ] Retest model eos2v11 post-refactorization
- [ ] If time, prep for Thursday presentation

**Comment by febielin** [comment]
- [ ] Ersilia inputs meeting 9 -10 AM
- [ ] eos2v11: work on failed action to upload onto dockerhub
- [ ] eos1af5: work on failed model test action–missing arg?
- [ ] eos7ack: IndexError

**Comment by GemmaTuron** [comment]
Hi @febielin !

I have suggested a way forward for the eos1af5, please see the issue. For model eos2v11, can you pin point the error to upload in Docker in the issue and we can discuss about it?
Is the eos7ack error the same we are having in eos46ev?

**Comment by febielin** [comment]
Hi @GemmaTuron,

Yes, I took your [suggestion](https://github.com/ersilia-os/eos1af5/issues/1#issuecomment-1611871839) and have constructing the path to the checkpoints dir, as opposed taking a third argument. However, I am still experiencing difficulties fetching models locally. Quite honestly, I am unsure if this is because there are fundamental issues in the code, or if my device just has trouble fetching local models. My suspicion is a combination of both (?) I am trying to work out the errors still before I submit another PR. 

For eos2v11, my understanding is that it is not able to upload to docker because of an error in the fetching process from github (`--from_github`). This does not surprise me because I have the same issue of not being able to fetch this model from github myself on CLI. I have come to a standstill trying to identify what's preventing a successful fetch. I may need additional support on this. 

I took a look at this [eos46ev issue ](https://github.com/ersilia-os/eos46ev/issues/5) and I have to say that the error seems very similar to the eos7ack error. I did a `git pull --rebase` to see if any of the changes made this morning would resolve my issue, but it's still producing the same IndexErrors and AssertionErrors. I am planning on discussing more on this during my presentation tomorrow. Hopefully we can generate some discussion around this during the meeting.

**Comment by febielin** [comment]
- [ ] Weekly Meeting 8-9 AM
- [ ] Ersilia Internship Meeting 9-9:30 AM
- [ ] eos1af5: make changes to code, and try ersilia shell in docker as a workaround for model testing
- [ ] eos7ack: try in codespaces, in docker container, --from_github
- [ ] eos2v11: check for issues regarding PID? if so, try to rerun the action workflow manually

**Comment by febielin** [comment]
Monday, July 3, 2023:

- [ ] Ersilia containers and environments meeting 8:30-9 AM
- [ ] eos1af5: Huddle with Simran
- [ ] eos24ci: cleanup and dockerization
- [ ] eos7ack: Assertion & IndexError

**Comment by GemmaTuron** [comment]
Hi @febielin 
Thanks for the explanation, I've left some comments on eos1af5 and eos24ci I hope this is helpful!

**Comment by febielin** [comment]
Tuesday, July 4, 2023:

- [ ] eos24ci: functionality + reformatting
- [ ] eos1af5: sort out packages and versions issue
- [ ] Weekly meeting 8-9 AM
- [ ] Intership meeting 10-10:30

**Comment by GemmaTuron** [comment]
Hi @febielin 
Thanks for the summary. Let's focus on closing the open issues before proceeding, let me know where are you at the end of the day today!

**Comment by febielin** [comment]
Wednesday, July 5:

- [ ] [eos3zur](https://github.com/ersilia-os/eos3zur/issues/5#event-9727353293): Model Testing
- [ ] eos24ci: rdkit issue
- [ ] eos1af5: sort out packages and versions issue
- [ ]  eos7ack: further testing

**Comment by GemmaTuron** [comment]
Hi @febielin 

Good work and perseverance, oftentime the fails are the most simple errors we have missed, like the {0} in the run.sh sorry I did not catch that either. The PRs are merged.

For eos1af5, you might be right, let's see if we merge the PR, please open it when ready: https://github.com/ersilia-os/eos1af5/pulls

I won't assign new models so we can focus on the remaining open ones!

**Comment by febielin** [comment]
Monday, July 10, 2023:

- [ ] eos2v11: model testing to confirm that it works
- [ ] eos24ci: ersilia install error
- [ ] eos8a4x: null error
- [ ] eos7ack: index error

**Comment by febielin** [comment]
Tuesday, July 11, 2023:

- [ ] eos4u6p: Model Testing
- [ ] eos8lok: clean up and dockerization
- [ ] Weekly meeting 8-9 AM
- [ ] Internship meeting 9-9:30 AM
- [ ] eos7ack: null and index errors

**Comment by GemmaTuron** [comment]
Hi @febielin 

the issue with Ersilia has been solved so it will probably work just fine on colab now. I've assigned you a new model in case you finish with eos8lok, I've left some comments on that model issue as well

**Comment by febielin** [comment]
Wednesday, July 12, 2023:

- [ ] Retest on Colab: eos4u6p, eos2mrz, eos8lok
- [ ] eos8lok: cleanup and dockerization
- [ ] eos30gr: cleanup and dockerization
- [ ] eos54ij: cleanup and dockerization

**Comment by febielin** [comment]
Thursday, July 13:

- [ ] eos30gr: investigate empty output
- [ ] eos54ij: investigate empty output

**Comment by GemmaTuron** [comment]
Hi @febielin 

Great explanations thanks. I've left comments on both eos30gr and eos54ij. I think retraining the eos30gr is a good option, check the training files to udnerstand the steps we took

**Comment by febielin** [comment]
Monday, July 17th:

- [ ] eos30gr: empty output error
- [ ] eos54ij: empty output error

**Comment by febielin** [comment]
Tuesday, July 18th, 2023:

- [ ] eos30gr: retraining
- [ ] Weekly meeting 8-9 AM
- [ ] Internship meeting 9-9:30 AM
- [ ] (maybe) Wait for Riley's assignments for test command

**Comment by febielin** [comment]
Thursday, July 20th, 2023:

- [ ] Lab Meeting 8-9 AM
- [ ] eos85a3: checkpoints refactoring
- [ ] eos43at: review Hellen's work, maybe change eos1af5 such that all molgrad models are consistent
- [ ] eos30gr: 
- [ ] test command: for consistency_check, if outputs are reported inconsistent, report the two outputs for easy comparison by contributor
- [ ] Review [Ersilia web documentation](https://ersilia.gitbook.io/ersilia-book/); look for areas that can be made more robust.

**Comment by febielin** [comment]
Monday, July 24th, 2023:

- [ ] eos30gr: next steps after retraining with lazy-qsar
- [ ] eos85a3: PR submitted, waiting on Ersilia 3.11 issue to see if it works
- [ ] Review [Ersilia web documentation](https://ersilia.gitbook.io/ersilia-book/)

**Comment by GemmaTuron** [comment]
Hi @febielin !

Thanks for the work, I won't assign new models until we are sure we have a good model for eos30gr, let's focus on training this model

**Comment by febielin** [comment]
Tuesday, July 25, 2023:

- [ ] [eos2ta5](https://github.com/ersilia-os/eos2ta5/issues/3#event-9908157465): new model testing
- [ ] [eos30gr](https://github.com/ersilia-os/eos30gr/issues/1#issuecomment-1644643956): model training
- [ ] test command: `bash run.sh `comparison with `ersilia run`

**Comment by febielin** [comment]
Monday, July 31:

- [ ] eos30gr: [Gemma's tips ](https://github.com/ersilia-os/eos30gr/issues/1#issuecomment-1658014845)
- [ ] eos4bj8: check server against model
- [ ] Test Module: bash run.sh
- [ ] Gitbook documentation

**Comment by febielin** [comment]
Wednesday, August 2, 2023:

- [ ] eos2mhp: model testing
- [ ] eos3xip: model testing
- [ ] eos9c7k: model testing (waiting for Riley's reformatting)
- [ ] eos85a3: troubleshoot Samuel's issue of null outputs - [link](https://github.com/ersilia-os/eos85a3/issues/4#issuecomment-1661088074)
- [ ] eos30gr: model retraining
- [ ] Lab Meeting 8-9 AM
- [ ] Internship Meeting 9-9:30 AM
- [ ] eos54ij: decide what to do with this model -- server is back up

**Comment by febielin** [comment]
Thursday, August 3, 2023:

- [ ] eos9c7k: model testing (waiting for Riley's reformatting)
- [ ] eos30gr: model training 
- [ ] eos54ij: decide what to do with this model -- server is back up
- [ ] retry eml_canonical runs for certain models: eos85a3, eos3xip, eos2mhp

**Comment by febielin** [comment]
Monday, August 7, 2023:

- [ ] eos9c7k: model testing
- [ ] eos30gr: final training
- [ ] Gitbook documentation
- [ ] Ersilia test: bash run.sh

**Comment by febielin** [comment]
Tuesday, August 8, 2023:

- [ ] eos2a9n: model testing
https://github.com/ersilia-os/eos2a9n/issues/3#issuecomment-1669724461
- [ ] Weekly Meeting 8-9 AM
- [ ] Internship Meeting 9-9:30 AM
- [ ] Gitbook documentation
- [ ] eos30gr: follow up on the issues
- [ ] Ersilia test: bash run.sh

**Comment by febielin** [comment]
Wednesday, August 9, 2023:

- [ ] eos30gr: remove conversion from .sdf --> .csv in main.py
- [ ] Test module: run.sh
- [ ] Gitbook Documentation

**Comment by febielin** [comment]
Hi @GemmaTuron,

Here are my updates for the test command. As a brief overview, we are working on getting the conda environment created and activated such that the model packages can be installed  and `bash run.sh` could be run within this new env. The main difficulty I am facing is getting this conda environment set up. I considered doing subprocesses for each step (conda create, conda activate, install packages, etc), but I believe these subprocesses execute in isolation so the results of one subprocess would not carry over. Looking through the existing ersilia code, I got the idea of creating and running bash script within this python script. 

I will point your attention to the bash script (towards the end), as I think this is most pressing. You'll see that I have the entire bash script planned out, but it has been commented out for the time being because it stalls forever. I decided to implement it line by line to figure out the point at which the code stalls. So far, I believe the code is stalling at the line `conda create ...`, although I am having difficulty understanding why. If I paste this same line of code into the command line, it works. It just doesn't work inside of this bash script. 

I also thought maybe it wasn't an issue of `conda create` specifically, but all conda commands. However, I tried replacing the conda create line with `conda env list` and `conda activate`, and I've verified that these indeed work. So the issue _is_ just in the conda env creation.

FYI, I've left some comments throughout the code to help you easily understand what I am doing. Some additional notes about the individual functions:
- _parse_dockerfile(): does precisely what the name suggests. It retrieves all the necessary installs in a list and adds the '-y' or '--quiet --no-input' flags so that user input will not be required
- default_env(), conda_prefix(), is_base() are all functions that I took from [conda.py](https://github.com/ersilia-os/ersilia/blob/3da1efc6b2201de1583cf2935a0792b72a89fbf5/ersilia/utils/conda.py#L4) since it accomplishes the same goal

```
    def _parse_dockerfile(self, temp_dir, pyversion):
        packages = set()
        prefix = "FROM bentoml/model-server:0.11.0-py"
        os.chdir(temp_dir) # navigate into cloned repo
        with open('Dockerfile', 'r') as dockerfile:
            lines = dockerfile.readlines()
            assert lines[0].startswith(prefix)
            pyversion[0] = lines[0][len(prefix):]
            lines_as_string = '\n'.join(lines)
            run_lines = re.findall(r'^\s*RUN\s+(.+)$', lines_as_string, re.MULTILINE)

        pip_prefix = "pip install"
        conda_prefix = "conda install"
        
        modified_run_lines = [] 
        for line in run_lines:
            if line.startswith(pip_prefix):
                line = line.replace(pip_prefix, "pip install --quiet --no-input")
            elif line.startswith(conda_prefix):
                line = line + " -y"
            else:
                print("not conda or pip")
            
            modified_run_lines.append(line) 

        return modified_run_lines  
    
    @staticmethod
    def default_env():
        if "CONDA_DEFAULT_ENV" in os.environ:
            return os.environ["CONDA_DEFAULT_ENV"]
        else:
            return BASE
    
    @staticmethod
    def conda_prefix(is_base):
        o = run_command_check_output("which conda").rstrip()
        if o:
            o = os.path.abspath(os.path.join(o, "..", ".."))
            return o
        if is_base:
            o = run_command_check_output("echo $CONDA_PREFIX").rstrip()
            return o
        else:
            o = run_command_check_output("echo $CONDA_PREFIX_1").rstrip()
            return o


    def is_base(self):
        default_env = self.default_env()
        if default_env == BASE:
            return True
        else:
            return False

    @throw_ersilia_exception
    def run_bash(self): 
        print("Cloning a temporary file and calculating model size...")
        
        current_dir = os.getcwd()

        # Create temp directory and clone model repo
        with tempfile.TemporaryDirectory() as temp_dir:
            repo_url = 'https://github.com/ersilia-os/{0}.git'.format(self.model_id) 
            try:
                subprocess.run(['git', 'clone', repo_url, temp_dir], check=True)
            except subprocess.CalledProcessError as e:
                print(f"Error while cloning the repository: {e}")

            # Print size of model
            self._set_model_size(temp_dir)
            size_kb = self.model_size / 1024
            size_mb = size_kb / 1024
            size_gb = size_mb / 1024
            print("\nModel Size:")
            print("KB:", size_kb)
            print("MB:", size_mb)
            print("GB:", size_gb)

            click.echo(BOLD + "\nRunning the model bash script..." + RESET)
            
            # halt this check if the run.sh file does not exist (e.g. eos3b5e)
            if not os.path.exists(os.path.join(temp_dir, "model/framework/run.sh")):
                print("Check halted: run.sh file does not exist.")
                return

            # Navigate into the temporary directory
            subdirectory_path = os.path.join(temp_dir, "model/framework")
            os.chdir(subdirectory_path)

            # Parse Dockerfiles
            pyversion = [0]
            packages = self._parse_dockerfile(temp_dir, pyversion)
            packages_string = '\n'.join(packages)
            pyversion[0] = pyversion[0][0] + '.' + pyversion[0][1:]

            try:
                # Create temp file
                with tempfile.NamedTemporaryFile(mode='w', delete=False) as temp_file:
                    temp_file_path = temp_file.name
                
                # Run bash script with specified args
                output_path = temp_file_path
                run_path = os.path.join(temp_dir, "model/framework/run.sh") 
                arg0 = os.path.join(temp_dir, "model/framework/")
                arg1 = os.path.join(current_dir, "ersilia/test/inputs/compound_singles.csv")      # input
                arg2 = output_path    

                tmp_script = os.path.join(temp_dir, "script.sh")
                
                ''' THIS IS THE FULL, PLANNED BASH SCRIPT
                bash_script += """
                source {6}/etc/profile.d/conda.sh
                conda create -n {0} python={1} -y
                conda activate {0} -y
                {2}
                result=$(bash run.sh {3} {4} {5})
                echo "Result of the command: $result"
                conda deactivate
                """.format(self.model_id, pyversion[0], packages_string, arg0, arg1, arg2, self.conda_prefix(True))
                '''

                # try the bash script line by line - TO FIGURE OUT WHERE IT STALLS
                # Currently, stalls at the conda create line
                bash_script = """
source {0}/etc/profile.d/conda.sh 
conda create -n {1} python={2} -y     
conda activate {1}
                """.format(self.conda_prefix(self.is_base()), self.model_id, pyversion[0])           

                with open(tmp_script, "w") as f:
                    f.write(bash_script)
                    
                # print contents of bash script to console
                with open(tmp_script, "r") as f:
                    tmp_script_contents = f.read()
                    print(tmp_script_contents)

                print("Executing the internal bash script...")
                
                run_command("bash {0}".format(tmp_script))

                #completed_process = subprocess.run(['bash', tmp_script], capture_output=True, text=True, check=True)
                #print(completed_process.stdout)


                # TODO: Remove this because this executes in the ersilia conda env, rather than the model conda env
                try:
                    subprocess.run(['bash', run_path, arg0, arg1, arg2,], check=True)
                except subprocess.CalledProcessError as e:
                    print(f"Error while running the bash script: {e}")

                with open(output_path, 'r') as temp_file:
                    output_contents = temp_file.read()

                print("Output contents:")
                print(output_contents)

            except Exception as e:
                    print(f"Error while creating or activating the conda environment: {e}")
```

**Comment by febielin** [comment]
Thursday, August 10, 2023:

- [ ] Lab Meeting 8-9 AM
- [ ] Internship Meeting 9-9:30 AM
- [ ] Gitbook Documentation
- [ ] Test Module: bash run.sh

**Comment by GemmaTuron** [comment]
Hi @febielin 

Good job, I think the test model is almost ready? the output could be saved to a temp directory probably? Once this is completed, I've assigned to you model `eos5y9f`, which we started to work on but never finished incorporating, let's see if we can tackle it this last week.

**Comment by febielin** [comment]
Monday, August 14, 2023:

- [ ] Test Module
- [ ] eos5y9f
- [ ] Prepare presentation for tomorrow

**Comment by febielin** [comment]
@GemmaTuron 

I am testing my changes for the test command, and I am finding that conda environments are not always created at fetch time. For example, here is my most recent fetch of eos92sw: [eos92sw_test.log](https://github.com/ersilia-os/ersilia/files/12353475/eos92sw_test.log)

As you can see, the bash run was successful in this situation, but upon closer inspection of the captured error log, I realized that it was operating from my ersilia conda environment (which happens to have scikit-learn version 1.0.2. This led me to check the fetch log to discover that no conda environment had been created. 

Why is this the case that the model conda environment is not always created? Also, would it be better to leave the conda create line to be certain?

**Comment by febielin** [comment]
Tuesday, August 15, 2023:

- [ ] Weekly Meeting 8-9 AM
- [ ] Internship Meeting 9-9:30 AM
- [ ] test command: bash run.sh
to do: remove conda create
- [ ] eos5y9f

**Comment by GemmaTuron** [comment]
Hi @febielin 

You are fetching without specifying from, so it is fetching from dockerhub first, this runs the image and will not create an environment that you can activate.
This is the discussion we were having yesterday. Since models will be fetched from docker (otherwise, for example, we run into Mac compatibility issues) maybe testing the run.sh command is not going to be possible, I think we decided to add a try - except where the run.sh test command is only used if the model is fetched from S3 or from GitHub? and then we do not need to clone the repo again or create the conda env, just use the ones that Ersilia has created. 
Does this make sense?

**Comment by febielin** [comment]
Wednesday, August 16, 2023:

- [ ] test module: test bash on several models
Models tested:
- [ ] eos5y9f: clean up and dockerization

Please merge this PR: https://github.com/ersilia-os/eos1af5/pull/11
These are changes that I made according to Hellen's findings.

**Comment by febielin** [comment]
> Hi Febie, sorry I missed the PR. It is now merged! Let me know if you need help with the test module, I'm around.

Thanks Gemma!

I'm currently working on the changes you proposed: first checking if the model is fetched from github/S3 before proceeding with the bash run. I am trying to search through the ersilia codebase for the syntax, but maybe you can help provide insight--how can I check if a model was fetched via github or S3? 

I see something in the code that looks like `self._s3`. Is this something I can use to check?

**Comment by GemmaTuron** [comment]
If the model is fetched from github or s3, it will download the whole repo on the eos/ folder, if it comes from docker it will just have placeholder folders there, so if we identify a file that is present when downloading from github or from s3 but not in the placeholder, that should work?
just guessing here!

**Comment by febielin** [comment]
> maybe even metadata.json in eos/dest/model_name

I agree! Let me test this out right now.

**Comment by febielin** [comment]
@GemmaTuron 

This approach works, but I realize that I already perform a check to see if the run.sh file exists. I think it would be smart to use this instead, because it will filter out:

(1) models that are not fetched via github/S3
(2) models that _are_ fetched via github/S3, but lack a run.sh file

**Comment by GemmaTuron** [comment]
> @GemmaTuron
> 
> This approach works, but I realize that I already perform a check to see if the run.sh file exists. I think it would be smart to use this instead, because it will filter out:
> 
> (1) models that are not fetched via github/S3 (2) models that _are_ fetched via github/S3, but lack a run.sh file (rendering this test useless)

yes, that makes sense!

**Comment by febielin** [comment]
@GemmaTuron 

The only problem I'm facing now is that I am unable to test it with this restriction. Both the fetch from s3 and fetch from github gives me the segmentation fault. I usually fetch all models from dockerhub.

I may need to recruit someone else to help me test this.

**Comment by GemmaTuron** [comment]
yes, that's true.
Could you update what you think is a working version on your fork and ask for help in the slack channel?

**Comment by febielin** [comment]
> yes, that's true. Could you update what you think is a working version on your fork and ask for help in the slack channel?

Hi @GemmaTuron,

Yes I have pushed all changes to my fork. It is hard for me to tell if there are still errors lurking around. But I will ask for help in the slack channel now.

**Comment by febielin** [comment]
Thursday, August 17, 2023:

- [ ] Test command: testing stage + minor changes
The test now uses _example_ to generate input, and it is placed into a temporary file with the header "smiles". The input will be processed twice, once by bash run and once by ersilia run. The two outputs will then be compared for consistency.
Recent push: https://github.com/febielin/ersilia.git
- [ ] Weekly Meeting 8-9 AM
- [ ] Ersilia internship meeting 9-9:30 AM
- [ ] eos5y9f: clean up and dockerization
Yesterday, I tried retraining the model with the provided data and notebooks from the publication, but I ran into an error involving `ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (4815352,) + inhomogeneous part.` Today, I tried following the provided jupyter notebook to use their pre-trained model instead. This also resulted in the ValueError. I tried a couple solutions I found online, such as downgrading numpy, but this did not work. It would be helpful for me to understand the root of this error because right now, neither pre-train or retrain options can get past this error.
